### DISCUSSION TITLE
What phones are compatible with SCREENSAVER xpdrip?

### DISCUSSION THREAD

**Discussion by pablodr350** [Navid200/xDrip#124] [General]
At this time, I know only Samsung phones have this option (Screen, Screensaver and selecting an app, Xdrip)

are there other phones where you can use the XDRIP screensaver when charging?

Thanks

**Comment by Navid200** [comment]
All as far as I know.
You just have to restart the phone after setting it up.
There are many xDrip settings that requires a restart.  This is one of them.

**Comment by pablodr350** [comment]
Screensaver isn´t avaliable in all phones, just tested in Xiaomi ie (for sure in Samsung, and I read about pixel, but not real option).
Screensaver isn´t a XDRIP option. XDRIP install service, but you need to select XDRIP in the Screensaver option on phone. If phone hasn´t the SCREENSAVER option, you can´t select it.

I would like to buy a phone with screensaver option enabled, and at this time only Samsung is option (and I would like to buy other phone due high price for samsung), but this option is mandatory to me (when I install phone in wireless charger on motorcycle or car, automatically shows the glucose value and graph, or in the night near to bed, charging wireless).

thanks

**Reply by Navid200** [reply]
I see the option on several Motorola phones and one Asus phone that I own.

Looking on Google, I see this for one particular xiaomi:
https://www.webcazine.com/10456/how-to-setup-screen-saver-on-xiaomi-redmi/

In this article, I see "Some, like EMUI or MIUI, limit the use of the screensaver on Android"   :    https://itigic.com/activate-and-configure-screensaver-on-android-phones/

I have no experience with Xiaomi.  So, you may have a point.

**Comment by pablodr350** [comment]
Thanks... what motorola phone? I read this option was removed from lastest updates, years ago,(on motorola-google phones). Asus did have the XDRIP screen saver option?

the link you put is from 2014 and maybe first xiaomi phone... for sure not avaliable. I have long time searching a phone actually with this option (Screensaver, and xdrip compatible). In pixel this options were, but was removed by google.

I think at this time, confirmed, only working in Samsung phones.

is present this option in your asus phone (you can select the screen protector by default, or an app, in this case XDRIP, this option appear after install xdrip? what model and version? motorola, what model?

I´m afraid samsung only option to use xdrip screensaver (not always on display, ie). Screensaver option enables when you are charging (only) and shows xdrip graph without user intervention.

Very annoying, need to buy a samsung phone to use the best option¡¡¡

Thanks
![SmartSelect_20211121-211503_Settings](https://user-images.githubusercontent.com/38194348/142777567-bc01956c-dd8a-4309-8075-c05cde07bc2d.jpg)

**Comment by Navid200** [comment]
Asus Zenfone 3 Zoom (Android 8):
![Screenshot_20211121-152215](https://user-images.githubusercontent.com/51497406/142778318-018134e1-c331-489d-b394-6a449ae466f8.jpg)

Moto Z3 play (Android 9)

Moto G Stylus (Android 11)

**Comment by pablodr350** [comment]
Exact, that´s the option.

But they are old phones (only new one is G Stylus if it´s 2021 model, actual for sale), as I said, android OS had this option enabled time ago but it was removed (by google), and I don´t know why (phones based in pure OS had it).

I would like to know, actual phones for sale, brand new, with this option enabled as Samsung (with SXX models, I don´t know if others Axx have it too).
Interested in ONE PLUS, or Xiaomi 11.... or similar medium-high performance options of users that bought it recently and using it.
Samsung is the only confirmed (in brand new and for sale now) phones with this option enabled (moto G Stylus if you confirm is 2021 model, actual for sale, too, I read in forums about it and was removed, but if you confirm it´s there, it´s nice)

thanks¡

**Comment by Navid200** [comment]
>  as I said, android OS had this option enabled time ago but it was removed (by google)

Is there an official article you can share declaring that Google has removed the feature?

**Comment by pablodr350** [comment]
Not official, but yes in some motorola review, when I was looking for information, that this feature was removed by google in the "pure" OS (not personalized with layouts) some brands use (present in previous versions, but some time ago, not present in updates).

And seems it´s true because I´m unable to found any actual phone (this year, to buy brand new) with this option, or some xdrip user with a brand new phone (brand and model please) using it. And I´m looking for time ago¡¡¡¡

I only want some option to buy a brand new phone in next months, actual model, with this option, and at this time, confirmed, only samsung SXX (I confirmed with S21) models. I don´t know if A series (samsung, have it, but I supposed it), but other brands, unable to confirm this option (xiaomi Note 8 pro, for sure haven´t, Note 3, too, no option...  mi 11, I don´t know). If some person is using Xdrip with a Xiaomi MI11 series, or a One Plus lastest versions, and can check if have this option (screensaver , and Xdrip app to select), I will thank you¡

Seems that now, it depends of manufacturer to enable the option or not. And seems, not is the most common. Unluckily.

Thanks

**Reply by chuckgit99** [reply]
One+8 5G has the option
![Screenshot_20211204-203918](https://user-images.githubusercontent.com/45901814/144730190-3ada71ee-2287-45b1-93a6-ece2e7e6bc73.jpg)

**Reply by pablodr350** [reply]
Thanks, great to know it¡¡¡ even 2 years old phone, it´s powerful (really I need this option and too, wireless charge, I think one plus 8 haven´t). But thanks¡¡¡¡ maybe new one plus models have it. They will be an option

**Reply by chuckgit99** [reply]
Unfortunately, no it doesn't have wireless charging.  Like you said, maybe the newer
One+ models do.

**Comment by Navid200** [comment]
> Not official, 

Can you provide any link, even unofficial, of anyone reporting this?

**Comment by pablodr350** [comment]
I will have to look for again about this option as I did some months ago when investigating about it.
Maybe xda-developer forum and some review about it (screensaver), I remember was talking about motorola device, this reference I read.

It would be easier to find an actual phone (actual is recent, for sale now, not old) phone with this option enabled :)

if some user have an actual phone (bought this year) can report model and brand and if it´s working¡¡¡

nobody is using Xdrip with recent phone? or maybe, people isn´t using this option (even if it´s there)? I need be in mistake to have other option (phone) to buy... (mainly medium-high devices, different to Samsung Sxx models)

Thanks

**Reply by roco5** [reply]
I have a pixel 4a and 5a and the screen saver is available as an android feature on both, but now xdrip is not in the selection options for screensaver.  Only clocks.

**Reply by Navid200** [reply]
Would you please show a screenshot of 4A?

**Reply by roco5** [reply]
https://photos.app.goo.gl/4quKdNzkAxrj9AfXA 

Sure -This is the 5a but it is the same as on my 4a.  Just no xdrip option under screensavers in android settings

![Image 2704](https://user-images.githubusercontent.com/6946920/145099666-26046a0d-ec63-46cb-96dd-2a07307086b5.jpg)
s.

**Reply by Navid200** [reply]
While you are editing the post, tap on the bottom border, it let's you choose file to attach.
Please do that so that everyone can see the image like the other images attached by others instead of linking to an external file server.

**Reply by roco5** [reply]
ok

**Reply by Navid200** [reply]
@roco5 Would you please use your 5a and go to the page you showed and tap on the "Current screen saver Clock"?  
Does it let you choose xDrip then?

**Comment by jredders** [comment]
Xdrip+ is in the list on my Pixel 5A running Android 12 with the December security patch when I click on "Current Screen Saver".
![Screenshot_20211210-092207](https://user-images.githubusercontent.com/9410439/145598378-8ed578d6-294a-4442-9c97-a8215a30cec8.jpg)

**Reply by Navid200** [reply]
Excellent.  Thank you for sharing.

**Comment by roco5** [comment]
Yep. Just got it too.

**Reply by Navid200** [reply]
Great

So, I don't think the capability is removed from new phones as was suggested.
We just needed to know how to use the phone.

Thanks to @jamorham for the solution.
